File: JS/21-30/e_22.js
function truncarDecimal(numero, cifrasDespuesComa) {
    var factor = Math.pow(10, cifrasDespuesComa);
    var resultado = Math.trunc(numero * factor) / factor;
    return resultado;
}

// Ejemplo de uso
var numeroOriginal = 342.9478274632;
var cifrasDespuesComa = 2;

var numeroTruncado = truncarDecimal(numeroOriginal, cifrasDespuesComa);

console.log('Número original:', numeroOriginal);
console.log('Número truncado a', cifrasDespuesComa, 'cifras después de la coma:', numeroTruncado);


// En este ejemplo, la función truncarDecimal multiplica el número original por 10 elevado a la potencia de las cifras deseadas después de la coma, 
// luego aplica Math.trunc() para eliminar la parte decimal,
// finalmente divide el resultado por el mismo factor para obtener el número truncado con la cantidad deseada de cifras después de la coma.
// El resultado se imprime en la consola.

File: Python/21-30/e_22.py
# Escribe un programa en python que permita truncar un número decimal a 2 cifras después de la coma.
# Por ejemplo, podemos usar el número decimal 342.9478274632


def truncar_decimal(numero, cifras_despues_coma):
    resultado = round(numero, cifras_despues_coma)
    return resultado

# Ejemplo de uso
numero_original = 342.9478274632
cifras_despues_coma = 2

numero_truncado = truncar_decimal(numero_original, cifras_despues_coma)

print(f'Número original: {numero_original}')
print(f'Número truncado a {cifras_despues_coma} cifras después de la coma: {numero_truncado}')


# Este programa define una función llamada truncar_decimal que toma un número y el número de cifras que quieres después de la coma. 
# Luego, utiliza la función round() para realizar el truncamiento y devuelve el resultado. 
# En el ejemplo de uso, se imprime el número original y el número truncado con 2 cifras después de la coma.